py4DSTEM/process/diffraction/crystal_bloch.py

```diff
@@ -397,16 +404,38 @@ def generate_dynamical_diffraction_pattern(
 
     # calculate the diffraction intensities for each thichness matrix
     # I = |psi|^2 ; psi = C @ E(z) @ C^-1 @ psi_0, where E(z) is the thickness matrix
-    intensities = [
-        np.abs(C @ (np.exp(2.0j * np.pi * z * gamma) * (C_inv @ psi_0))) ** 2
-        for z in np.atleast_1d(thickness)
-    ]
+    if return_complex: 
+        # not an intensity but ... yolo 
+        # TODO refactor logic and fix the naming 
+        intensities = [
+            C @ (np.exp(2.0j * np.pi * z * gamma) * (C_inv @ psi_0))
+            for z in np.atleast_1d(thickness)
+        ]
+    else: 
+        intensities = [
+            np.abs(C @ (np.exp(2.0j * np.pi * z * gamma) * (C_inv @ psi_0))) ** 2
+            for z in np.atleast_1d(thickness)
+        ]
 
     # make new pointlists for each thickness case and copy intensities
     pls = []
     for i in range(len(intensities)):
         newpl = beams.copy()
-        newpl.data["intensity"] = intensities[i]
+        # if we return the complex amplitude we need to change the dtype of the struct array
+        # TODO move this if statement outside of the loop and rename intensities) better 
+        if return_complex: 
+            newpl.data = newpl.data.astype(dtype=[
```

Comment:
This method of changing the datatype of the PointList is not ideal. Doing it this way hard-codes assumptions about the datastructure of `beams`, which the code currently does not require. This also breaks a lot of other stuff, like the plotting code, that looks for the `'intensity'` field. I think you should actually add a new field for the amplitude.

Reply:
updated so  that `return_complex = True` returns points list with dynamical values for `intensity` and `amplitude`. Plotting functions work correctly with pointlists generated with `return_complex = True`